components/cdc/src/endpoint.rs

```diff
@@ -439,7 +439,14 @@ impl<T: 'static + RaftStoreRouter> Endpoint<T> {
             SignificantMsg::CaptureChange {
                 cmd: change_cmd,
                 region_epoch: request.take_region_epoch(),
-                callback: Callback::Read(cb),
+                callback: Callback::Read(Box::new(move |resp| {
+                    downstream_state.compare_and_swap(
```

Comment:
When does the compare fails? Can we assert the value if the compare is not supposed to fail?